### ISSUE TITLE
Crashes when epr_socket_id set as template in request routine 

### ISSUE THREAD

**Issue by ThomasRoyBeauchamp** [QuTech-Delft/qoala-sim#33]
When trying to set the epr_socket_id as a template/program input, this results in the following error being thrown: 

```INFO:0.0 ns:Stack.CpuEdfScheduler_CpuDriver(alice_cpu):executing task PreCallTask(pid=0, tid=0)
INFO:0.0 ns:Stack.HostProcessor(alice_host_protocol_HostProcessor):executing request routine RequestRoutine(name='req', request=QoalaRequest(name='req', remote_id=Template(name='bob_id'), epr_socket_id=Template(name='socket'), num_pairs=1, virt_ids=RequestVirtIdMapping(typ=<VirtIdMappingType.EQUAL: 0>, single_value=0, custom_values=None), timeout=1000, fidelity=1.0, typ=<EprType.MEASURE_DIRECTLY: 1>, role=<EprRole.CREATE: 0>, window=None), return_vars=['m0'], callback_type=<CallbackType.WAIT_ALL: 1>, callback=None)
INFO:0.0 ns:Stack.CpuEdfScheduler_CpuDriver(bob_cpu):executing task PreCallTask(pid=0, tid=0)
INFO:0.0 ns:Stack.HostProcessor(bob_host_protocol_HostProcessor):executing request routine RequestRoutine(name='req', request=QoalaRequest(name='req', remote_id=Template(name='alice_id'), epr_socket_id=0, num_pairs=1, virt_ids=RequestVirtIdMapping(typ=<VirtIdMappingType.EQUAL: 0>, single_value=0, custom_values=None), timeout=1000, fidelity=1.0, typ=<EprType.MEASURE_DIRECTLY: 1>, role=<EprRole.RECEIVE: 1>, window=None), return_vars=['m0'], callback_type=<CallbackType.WAIT_ALL: 1>, callback=None)
INFO:0.0 ns:Stack.CpuEdfScheduler_CpuDriver(bob_cpu):finished task PreCallTask(pid=0, tid=0)
INFO:0.0 ns:Stack.CpuEdfScheduler_CpuDriver(alice_cpu):finished task PreCallTask(pid=0, tid=0)
I need to know my socket so I sure hope I have been instantiated
Traceback (most recent call last):
  File "/tmp/pycharm_project_633/baseline_qkd.py", line 161, in <module>
    app_result = run_network_schedule(schedule=network_scheduler.computed_schedule, network=network)
  File "/tmp/pycharm_project_633/Scheduler_Framework/simulation/multiple_end_nodes.py", line 181, in run_network_schedule
    ns.sim_run()
  File "/home/thomas/.virtualenvs/Network-Architecture-Evaluation/lib/python3.10/site-packages/netsquid/util/simtools.py", line 278, in sim_run
    _simengine.run()
  File "pydynaa/core.pyx", line 1466, in pydynaa.core.SimulationEngine.run
  File "pydynaa/core.pyx", line 172, in pydynaa.core._call_obj_expr
  File "pydynaa/core.pyx", line 169, in pydynaa.core._call_obj_expr
  File "netsquid/protocols/protocol.pyx", line 290, in netsquid.protocols.protocol.Protocol._expression_callback
  File "netsquid/protocols/protocol.pyx", line 301, in netsquid.protocols.protocol.Protocol._generator_step
  File "/home/thomas/.virtualenvs/Network-Architecture-Evaluation/lib/python3.10/site-packages/qoala/sim/scheduler.py", line 1136, in run
    status = self.update_status()
  File "/home/thomas/.virtualenvs/Network-Architecture-Evaluation/lib/python3.10/site-packages/qoala/sim/scheduler.py", line 1053, in update_status
    bin = self.timebin_for_task(e)
  File "/home/thomas/.virtualenvs/Network-Architecture-Evaluation/lib/python3.10/site-packages/qoala/sim/scheduler.py", line 923, in timebin_for_task
    epr_sck = process.epr_sockets[request.epr_socket_id]
KeyError: Template(name='socket')
```

It appears to be caused by the QoalaRequest object not having been instantiated before this, and therefore the socket_id has not been replaced with the input value. In particular, this occurs (I believe) when the scheduler updates itself after completing the PreCallTask for this RR. As far as I can tell, the RR is only instantiated once the [Multi/Single]PairTask object is scheduled for execution, which is after this scheduler update.

**Comment by bvdvecht** [comment]
Should be fixed by #35 . @ThomasRoyBeauchamp Could you check whether this is the case?

**Comment by ThomasRoyBeauchamp** [comment]
Checked and does appear to be the case, working fine with a quick test
